tools/toolchain/optimized_clang.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash -uex
+
+if [[ -z "${CLANG_BUILD}" ]]; then
+    echo "CLANG_BUILD not specified"
+    exit 1
+elif [[ "${CLANG_BUILD}" = "SKIP" ]]; then
+    echo "Skip building optimized clang"
+    exit 0
+elif [[ "${CLANG_BUILD}" = "INSTALL" ]]; then
+    echo "build and install optimized clang"
+elif [[ "${CLANG_BUILD}" = "INSTALL_FROM" ]]; then
+    echo "install prebuilt optimized clang"
+else
+    echo "Not sure what to do with ${CLANG_BUILD}"
+    exit 1
+fi
```

Comment:
nit, could use case-esac for dispatching based on the value of `CLANG_BUILD` for better readability.